packages/loader/container-loader/src/loader.ts

```diff
@@ -138,7 +163,7 @@ export interface ILoaderProps {
      * The code loader handles loading the necessary code
      * for running a container once it is loaded.
      */
-    readonly codeLoader: ICodeLoader;
+    readonly codeLoader: ICodeDetailsLoader;
```

Comment:
this should remain back compact and continue to support ICodeLoader

Reply:
i guess it does remain back compat, but i think i'd rather have the new interface only returning `IFluidModuleWithDetails` and then use a union here. the point being to move people away from the old interface, and not have to support the old interface forever, which we know causes problems in the upgrade flow

Reply:
Thanks to Typescript magic this solution is back compat :). Will refactor it as per the feedback. Meanwhile, should we mark the original `ICodeLoader` as `@deprecated`?

Reply:
Refactored.